kv/dist_sender.go

```diff
@@ -520,6 +526,50 @@ func (ds *DistSender) getDescriptors(call proto.Call) (*proto.RangeDescriptor, *
 	return desc, descNext, nil
 }
 
+// getDescriptorsForReverseScan takes a call and looks up the corresponding range
+// descriptors associated with it. First, the range descriptor for
+// call.Args.EndKey is looked up. If call.Args.Key less than the
+// returned descriptor, the next descriptor is obtained as well.
+func (ds *DistSender) getDescriptorsForReverseScan(call proto.Call, options lookupOptions) (*proto.RangeDescriptor, *proto.RangeDescriptor, error) {
```

Comment:
I think you should remove this method entirely. Too much duplicated code. Just set up an if/else at the top of getDescriptors and create different closures for checking whether another range descriptor is necessary and which key to query:

    needAnotherDescriptorFn := func(desc *proto.RangeDescriptor) bool { return call.Args.Header().Key.Less(desc.StartKey) }
    nextKeyFn := func(desc *proto.RangeDescriptor) proto.Key { return desc.EndKey }

vs.

    needAnotherDescriptorFn := func(desc *proto.RangeDescriptor) bool { return desc.EndKey.Less(call.Args.Header().EndKey) }
    nextKeyFn := func(desc *proto.RangeDescriptor) proto.Key { return desc.StartKey.Prev() }